health/health.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package health
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vechain/thor/v2/chain"
+	"github.com/vechain/thor/v2/comm"
+	"github.com/vechain/thor/v2/thor"
+)
+
+type BlockIngestion struct {
+	ID        *thor.Bytes32 `json:"id"`
+	Timestamp *time.Time    `json:"timestamp"`
+}
+
+type Status struct {
+	Healthy           bool            `json:"healthy"`
+	BlockIngestion    *BlockIngestion `json:"blockIngestion"`
+	ChainBootstrapped bool            `json:"chainBootstrapped"`
+	PeerCount         int             `json:"peerCount"`
+}
+
+type Health struct {
+	lock              sync.RWMutex
+	timeBetweenBlocks time.Duration
+	repo              *chain.Repository
+	p2p               *comm.Communicator
+}
+
+const delayBuffer = 5 * time.Second
+
+func NewSolo(timeBetweenBlocks time.Duration) *Health {
+	return &Health{
+		timeBetweenBlocks: timeBetweenBlocks + delayBuffer,
+		// there is no bootstrap in solo mode
+	}
+}
+
+func New(repo *chain.Repository, p2p *comm.Communicator, timeBetweenBlocks time.Duration) *Health {
+	return &Health{
+		repo:              repo,
+		timeBetweenBlocks: timeBetweenBlocks + delayBuffer,
+		p2p:               p2p,
+	}
+}
```

Comment:
I refactored the health service to accept components and work in a pull fashion.
I think it looks better, thanks guys 🙏